src/solver/highlevel.jl

```diff
@@ -61,3 +61,28 @@ function Base.show(io::IO, F::CuQR)
     println(io)
     show(io, F.R)
 end
+
+# Singular Value Decomposition
+
+struct CuSVD{T,Tr,M<:AbstractArray{T}} <: LinearAlgebra.Factorization{T}
+    U::M
```

Comment:
I think it's fine to make these `CuMatrix{T}` to simplify the type. It's unlikely that it'll be anything else for `CuiSVD`.

Reply:
That was my thinking initially, but noticed the `CuQR` and `CuQRPacked` are following the `AbstractMatrix` pattern. Perhaps we should change it there as well.

Reply:
Fixed in last commit.

Reply:
> Fixed in last commit.

It doesn't seems so. The `struct` hasn't changed.

Reply:
It has change in the commit below. :-)  
https://github.com/JuliaGPU/CuArrays.jl/pull/183/commits/22fe25094b51e96d928c568680de1472121e07a4

Reply:
I think we are talking past each other. Either that or I don't see the same diff as you. That commit doesn't change the struct.

Reply:
@andreasnoack 
You are absolutely right. Apologies. I missed that bit. 
Created a new pull request, since this one was merged already. 
https://github.com/JuliaGPU/CuArrays.jl/pull/187